examples/src/main/java/com/mapbox/navigation/examples/ui/BuildingExtrusionHighlightActivity.kt

```diff
@@ -99,51 +103,99 @@ class BuildingExtrusionActivity : AppCompatActivity(), OnNavigationReadyCallback
+                adjust_visibility_fab.setOnClickListener {
+                    mapboxMap.getStyle {
+                        val currentlyVisible = it.getLayer(BuildingExtrusionHighlightLayer.BUILDING_HIGHLIGHTED_EXTRUSION_LAYER_ID)
+                                ?.visibility?.value.equals(Property.VISIBLE)
+                        buildingExtrusionHighlightLayer.updateVisibility(!currentlyVisible)
                     }
-                    buildingExtrusionLayer.opacity = opacityList[adjustExtrusionsStyleButtonIndex]
-                    buildingExtrusionLayer.color = colorList[adjustExtrusionsStyleButtonIndex]
-                    adjustExtrusionsStyleButtonIndex++
                 }
             }
         }
     }
 
-    override fun onRouteProgressChanged(routeProgress: RouteProgress) {
-        // todo
+    override fun onNextRouteLegStart(routeLegProgress: RouteLegProgress) {
+        // Not needed in this example
+    }
+
+    override fun onFinalDestinationArrival(routeProgress: RouteProgress) {
+        mapboxMap.easeCamera(CameraUpdateFactory.zoomTo(18.0), 1800)
+
+        mapboxMap.getStyle {
+
+            /**
+             * Uncomment this commented out code and run the app if you want to add building
+             * extrusions to the map before the device arrives to to the final destination.
+             */
+            /*
+            val buildingFootprintVectorSource = VectorSource(
+                    "building-vector-source", "mapbox://mapbox.mapbox-streets-v8")
+            it.addSource(buildingFootprintVectorSource)
+
+            val fillLayer = FillExtrusionLayer("fill-building-test","building-vector-source")
+            fillLayer.sourceLayer = "building"
+            fillLayer.withProperties(
+                    fillExtrusionColor(Color.GRAY),
+                    fillExtrusionOpacity(.7f),
+                    fillExtrusionHeight(get("height"))
+            )
+            it.addLayer(fillLayer)
+            */
+
+            /**
+             * Set the [LatLng] to be used by the [BuildingFootprintHighlightLayer].
+             * The footprint or extrusion that's shown is whatever is a maximum of 1 meter
+             * away from the query [LatLng].
+             *
+             * The LatLng passed through below is different than the coordinate used as the
+             * final destination coordinate in this example's [DirectionsRoute].
+             *
+             * This LatLng should be set before the visibility of either the extrusion
+             * or footprint is set to true.
+             */
+            buildingExtrusionHighlightLayer.queryLatLng = highlightQueryLatLng
+
+            buildingExtrusionHighlightLayer.updateVisibility(true)
+
+            // Click on a building footprint to move the highlighted footprint
+            mapboxMap.addOnMapClickListener {
+                buildingExtrusionHighlightLayer.queryLatLng = it
+                true
+            }
+        }
```

Comment:
In this comment can we also mention how do we get this `highlightQueryLatLng` since here it is just harcoded, but is not going to be the case in general.